Shift3/boilerplate-client-angular
# chore, fix(angular, package, yarn): configure Transloco to work with AOT

## Changes

- [x] Add `@ngneat/transloco-optimize` as a dev dependency.
- [x] Explicitly enable the aot flag for production builds.
- [x] Add `transloco:optimize` script and use in build and deploy scripts.

## Purpose

When compiling for production compiles for AOT, the project cannot find the language JSON files. This results in showing the JSON keys instead of values:

![image](https://user-images.githubusercontent.com/488032/124843504-eb1a3480-df46-11eb-8b2c-82e3909bfddb.png)

## Pre-Testing TODOs

_What needs to be done before testing._

  1. Run `yarn install` to install the new `@ngneat/transloco-optimize` dependency.

## Testing Steps

#### If you are not a member of this project, _skip this step_

_How do the users test this change?_

  1. Run `ng serve --prod` locally OR run `yarn run build:prod` and then run `http-server dist/boilerplate-client-angular` after installing [http-server](https://www.npmjs.com/package/http-server).
  2. Load the webapp in a browser.
  3. Confirm that the language values show, not the language keys
  4. Try changing languages.
  5. Confirm that the language changes to the new selected language successfully.

## Learning

https://ngneat.github.io/transloco/docs/tools/optimize/
https://github.com/Shift3/bakersfield-goals/pull/87

Closes #333.
